scrape/scrape.go

```diff
@@ -414,21 +425,41 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 
 	sp.targetMtx.Lock()
 	var all []*Target
-	sp.droppedTargets = []*Target{}
+
+	// Clean up old targets from drop cache.
+	for k, t := range sp.droppedTargets {
+		if time.Since(t.dropTime) > droppedTargetTTL {
+			delete(sp.droppedTargets, k)
+		}
+	}
+	
 	for _, tg := range tgs {
+		tgHash := uint64(tg.Labels.FastFingerprint())
```

Comment:
FNV is not a safe fingerprint to use. Convert it to the type of label we use in Prometheus, and then use the Hash function there.